# Exercice : Analyse des données de résistance du béton avec des listes

### Lecture des données

Le cellule suivante lit les données de résistance du béton et les stocke dans une liste de listes. Chaque sous-liste contient les données d'une ligne du fichier.


In [6]:
file_path = "./resources/Concrete_Data_Yeh/Concrete_Data_Yeh.csv"

# lecture du fichier
with open(file_path, 'r') as file:
    list_tableau = []
    for line_id, line_content in enumerate(file):
        if line_id > 0:
            elements = line_content.strip().split(',')
            list_tableau.append(elements)
list_tableau

[['540', '0', '0', '162', '2.5', '1040', '676', '28', '79.99'],
 ['540', '0', '0', '162', '2.5', '1055', '676', '28', '61.89'],
 ['332.5', '142.5', '0', '228', '0', '932', '594', '270', '40.27'],
 ['332.5', '142.5', '0', '228', '0', '932', '594', '365', '41.05'],
 ['198.6', '132.4', '0', '192', '0', '978.4', '825.5', '360', '44.3'],
 ['266', '114', '0', '228', '0', '932', '670', '90', '47.03'],
 ['380', '95', '0', '228', '0', '932', '594', '365', '43.7'],
 ['380', '95', '0', '228', '0', '932', '594', '28', '36.45'],
 ['266', '114', '0', '228', '0', '932', '670', '28', '45.85'],
 ['475', '0', '0', '228', '0', '932', '594', '28', '39.29'],
 ['198.6', '132.4', '0', '192', '0', '978.4', '825.5', '90', '38.07'],
 ['198.6', '132.4', '0', '192', '0', '978.4', '825.5', '28', '28.02'],
 ['427.5', '47.5', '0', '228', '0', '932', '594', '270', '43.01'],
 ['190', '190', '0', '228', '0', '932', '670', '90', '42.33'],
 ['304', '76', '0', '228', '0', '932', '670', '28', '47.81'],
 ['380', '0', '0', '

### Compter le nombre de lignes dans le fichier

In [ ]:
# todo

### Compter les nombres d'élément par ligne dans le fichier.


In [ ]:
# todo

### Afficher uniquement les 3 premières lignes du fichier.

In [11]:
# todo

### Afficher uniquement les 3 dernières lignes du fichier.

In [ ]:
# todo

### Afficher uniquement que les lignes impaires du fichier.

In [ ]:
## todo

### Sélection d'un élément dans la liste

Selectionner la contenu en eau (en g par m3 de mélange) pour un ciment de 427.5 kg par m3 de mélange

In [13]:
# todo

### Nouvelle liste

Créer une nouvelle list "new_list" avec uniquement que la moitié des lignes du tableau

In [10]:
# todo

Créer une nouvelle "new_list_inverse" avec les lignes dans l'ordre inverse

In [ ]:
# todo